tests/Recovery.test.ts

```diff
@@ -151,177 +151,117 @@ describe ('Test of Recovery', () =>
-                        .directory("block")
-                        .addSearch("block_height", idx);
-
-                    let response = await client.get(uri.toString());
-                    assert.strictEqual(response.status, 200);
-                    assert.strictEqual(response.data.height, idx);
-                }
-
-                doneIt();
-
-            }, 800);
-        })();
-    });
-
-    it ('Test for ignoring already wrote block data', (doneIt: () => void) =>
-    {
-        (async () =>
+        // Verifies that all sent blocks are wrote
+        for (let idx = 0; idx <= 8; idx++)
         {
-            agora_node.delay = 100;
-
             let uri = URI(stoa_addr)
-                .directory("block_externalized");
-
-            let url = uri.toString();
-
-            await client.post(url, {block: recovery_sample_data[0]});
+                .directory("block")
+                .addSearch("block_height", idx);
 
-            // Delay to wait for data added to the pool to be processed.
-            setTimeout(() =>
-            {
-                client.post(url, {block: recovery_sample_data[1]});
-            }, 15);
+            let response = await client.get(uri.toString());
+            assert.strictEqual(response.status, 200);
+            assert.strictEqual(response.data.height, idx);
+        }
+    });
 
-            // Blocks 2 is recovered, Block 3 is saved
-            setTimeout(() =>
-            {
-                client.post(url, {block: recovery_sample_data[3]});
-            }, 30);
+    it ('Test for ignoring already wrote block data', async () =>
+    {
+        agora_node.delay = 100;
 
-            // Make sure Block 4 arrives during the saving of Block 2, and 3.
-            // 100ms - 30ms < 100ms(agora_node.delay)
-            setTimeout(() =>
-            {
-                client.post(url, {block: recovery_sample_data[4]});
-            }, 100);
+        let uri = URI(stoa_addr)
+            .directory("block_externalized");
 
-            // Block 3 is ignored.
-            // If Block 3 was not ignored and attempted to write
-            // to the database, an error would occur.
-            setTimeout(() =>
-            {
-                client.post(url, {block: recovery_sample_data[3]});
-            }, 130);
+        let url = uri.toString();
 
-            setTimeout(async () =>
-            {
-                // Verifies that all sent blocks are wrote
-                for (let idx = 0; idx <= 4; idx++)
-                {
-                    let uri = URI(stoa_addr)
-                        .directory("block")
-                        .addSearch("block_height", idx);
+        await client.post(url, {block: recovery_sample_data[0]});
+        await client.post(url, {block: recovery_sample_data[1]});
+        // Blocks 2 is recovered, Block 3 is saved
+        await client.post(url, {block: recovery_sample_data[3]});
+        await client.post(url, {block: recovery_sample_data[4]});
```

Comment:
The test code order has been changed?

Reply:
The source code below is the original code.
The code was simplified using the function delay(), and the github may not be detected due to a large amount of variation
``` TypeScript
        (async () =>
        {
            let uri = URI(stoa_addr)
                .directory("block_externalized");

            let url = uri.toString();

            await client.post(url, {block: recovery_sample_data[2]});

            setTimeout(() =>
            {
                client.post(url, {block: recovery_sample_data[4]});
            }, 15);

            setTimeout(() =>
            {
                client.post(url, {block: recovery_sample_data[6]});
            }, 30);

            setTimeout(() =>
            {
                client.post(url, {block: recovery_sample_data[8]});
            }, 45);

            setTimeout(async () =>
            {
                // Verifies that all sent blocks are wrote
                for (let idx = 0; idx <= 8; idx++)
                {
                    let uri = URI(stoa_addr)
                        .directory("block")
                        .addSearch("block_height", idx);

                    let response = await client.get(uri.toString());
                    assert.strictEqual(response.status, 200);
                    assert.strictEqual(response.data.height, idx);
                }

                doneIt();

            }, 800);
        })();
```

Reply:
Visual Studio Code detects well.